File: tree-walk/go/internal/lox/clock_native_function.go
package lox

import "time"

type clockNativeFunction struct{}

func newClockNativeFunction() clockNativeFunction {
	return clockNativeFunction{}
}

func (c clockNativeFunction) arity() int {
	return 0
}

func (c clockNativeFunction) call(interpreter *Interpreter, arguments []any) any {
	return time.Now().UnixMilli()
}

func (c clockNativeFunction) String() string {
	return "<native fn>"
}

File: tree-walk/cs/Lox/Interpreter/NativeFunctions/ClockNativeFunction.cs
﻿using System;

namespace Lox.Interpreter.Callables;

internal class ClockNativeFunction : ILoxCallable
{
    public int Arity()
    {
        return 0;
    }

    public object? Call(Interpreter interpreter, List<object> arguments)
    {
        return (double)DateTimeOffset.Now.ToUnixTimeMilliseconds() / 1000.0;
    }

    public override string ToString()
    {
        return "<native fn>";
    }
}

